dds/idl/typeobject_generator.cpp

```diff
@@ -519,17 +519,70 @@ operator<<(std::ostream& out, const OpenDDS::XTypes::TypeIdentifier& ti)
   return out;
 }
 
+namespace {
+template<typename T>
+std::string getTypeName();
+
+template<>
+std::string getTypeName<OPENDDS_STRING>()
+{
+  return "OPENDDS_STRING";
+}
```

Comment:
Please use `OpenDDS::DCPS::String` instead of `OPENDDS_STRING`.

Reply:
Isn't `OPENDDS_STRING` an alias to `OpenDDS::DCPS::String`?

Reply:
Yes, but I'm aiming to remove `OPENDDS_STRING` as part of OPENDDS-248, so it'd like to prevent any new usage of them.